src/v/container/include/container/json.h

```diff
@@ -15,10 +15,9 @@
 
 namespace json {
 
-template<typename T, size_t max_fragment_size>
+template<typename Buffer, typename T, size_t max_fragment_size>
```

Comment:
> Would it make sense to use `chunked_buffer` by default and drop all uses of `json::StringBuffer`? If not, I'm wondering where would be a good place to put a doc comment that explains when to use `json::StringBuffer` vs `chunked_buffer`.

This would be a very large change, I've added [CORE-5101]